tests/TestHarness/Node.py

```diff
@@ -550,6 +550,38 @@ def findInLog(self, searchStr):
                         return True
         return False
 
+    # verify only one or two 'Starting block' per block number unless block is restarted
+    def verifyStartingBlockMessages(self):
+        dataDir=Utils.getNodeDataDir(self.nodeId)
+        files=Node.findStderrFiles(dataDir)
+        for f in files:
+            blockNumbers = set()
+            duplicateBlockNumbers = set()
+            duplicatesFound = False
+            lastRestartBlockNum = 0
+
+            with open(f, 'r') as file:
+                for line in file:
+                    match = re.match(r".*Restarting exhausted speculative block #(\d+)", line)
+                    if match:
+                        lastRestartBlockNum = match.group(1)
+                    if re.match(r".*unlinkable_block_exception", line):
+                        lastRestartBlockNum = blockNumber
+                    match = re.match(r".*Starting block #(\d+)", line)
+                    if match:
+                        blockNumber = match.group(1)
+                        if blockNumber != lastRestartBlockNum and blockNumber in duplicateBlockNumbers:
+                            print(f"Duplicate Staring block found: {blockNumber} in {f}")
+                            duplicatesFound = True
+                        if blockNumber != lastRestartBlockNum and blockNumber in blockNumbers:
+                            duplicateBlockNumbers.add(blockNumber)
+                        blockNumbers.add(blockNumber)
+
+            if duplicatesFound:
```

Comment:
What happens if the same Start Block is spread across two stderr files?

Reply:
This `if` is inside the loop over the files. The first file that is found with a duplicate the verify stops. Might be better to scan all files for reporting. I'll make that change.